gadget/layout.go

```diff
@@ -30,15 +30,28 @@ import (
 	"github.com/snapcore/snapd/kernel"
 )
 
+// LayoutOptions defines the options to layout a given volume.
+type LayoutOptions struct {
```

Comment:
I don't have any immediate suggestion, but this is a bit odd for an Options as it's all directories. LayoutUnpackedSnaps ?

Reply:
otoh now that we have Options we should move IgnoreContent and SkipResolveContent here?  they are not really Constraints

Reply:
I created a followup in https://github.com/snapcore/snapd/compare/master...mvo5:gadget-refactor-2?expand=1 (see https://github.com/snapcore/snapd/commit/7249ec5e5452cbf06ffba97f62aaf407f0c3d5a2) I can cherry pick it here, it adds quite a bit of extra churn unfortunately.

Reply:
@mvo5 I see, I forgot about DefaultConstraints, so no, that gets weird. The better idea is probably to move the flags to Options but keep a reduced Constraints

Reply:
NIce, thank you! That makes a lot of sense and is a relatively small diff. Pushed the (small) update :)